py2exe/mf3.py

```diff
@@ -670,21 +671,39 @@ def __init__(self, loader, name, optimize):
             except AttributeError:
                 pass
 
+    @property
+    def __compiled_file__(self):
+        """Gets the path for the module that will be used as file name at compilation time."""
+        try:
+            source = self.__source__
+        except:
+            source = None
+        if source is None:
+            if hasattr(self, __file__):
+                return self.__file__
+            raise RuntimeError("getting compiled file from %r" % self) from None

```

Comment:
When is `source` expected to be `None`? Would not this code raise a `RuntimeError` in all those cases?

Reply:
Source is None when trying to load a module for which only the byte compiled file is available. IN that case, the compiled file name is just the current name of the file.
Having said that, the __compiled_file__ property is currently not used in a situation where there is no source. I just wanted to provide a fallback for the property in a future case where it is might be fetched on a module when there is no source.